pyglider/slocum.py

```diff
@@ -807,9 +811,35 @@ def binary_to_timeseries(indir, cachedir, outdir, deploymentyaml, *,
     outdir : string
         Directory to put the merged timeseries files.
 
-    deploymentyaml : str
+    deploymentyaml : string
         YAML text file with deployment information for this glider.
 
+    search : string
+        todo - dbdreader
+
+    fnamesuffix : string
+        Suffix for the output timeseries file
+
+    time_base : string
+        The parameter (sensor name) to be used as the time base by the
+        dbdreader MultiDBD.get_sync() method. Note that this means that all
+        parameters will be interpolated onto this time base.
+
+        If this value is 'union', then the processing is handled differently,
+        to allow for 'unioning' the engineering and science timeseries. This
+        may be useful if for instance you want a full time series, and science
+        variables are only sampled on dives.
+
+        For a value of 'union', the dbdreader MultiDBD.get() method is used
+        rather than get_sync to read the parameters specified in
+        deploymentyaml. The argument return_nans (of MultiDBD.get()) is set to
+        True, so that there are two 'time bases' for the extracted data: one
+        for engineering variables (from m_present_time), and one for science
+        variables (from sci_m_present_time). These times are rounded to the
+        nearest second, and then merged. These values are the time index of
+        the output file. In this case, only the engineering variables (e.g.,
+        lat/lon, pitch, roll, m_depth) are interpolated.
```

Comment:
```suggestion
        If this value is 'union', then the processing is handled differently,
        to allow for 'unioning' the engineering and science timeseries. This
        may be useful if for instance you want a full time series, and science
        variables are only sampled on dives.

        For a value of 'union', the dbdreader MultiDBD.get() method is used
        rather than get_sync to read the parameters specified in
        deploymentyaml. The argument return_nans (of MultiDBD.get()) is set to
        True, so that there are two 'time bases' for the extracted data: one
        for engineering variables (from m_present_time), and one for science
        variables (from sci_m_present_time). These times are rounded to the
        nearest second, and then merged. These values are the time index of
        the output file. In this case, only the engineering variables (e.g.,
        lat/lon, pitch, roll, m_depth) are interpolated.
```

This isn't super clear, and not what I understood was going to happen here.  If there are two science sensors on different time bases, I thought both times were going to be logged and NaN inserted for the sensors on a different time base.

Not clear on the "rounding to nearest second part".  Surely there are plenty of sensors sampled faster than 1 Hz that this will be a bad assumption for.  I'm not clear why you are doing this.

Why would the engineering sensors be treated differently than a mismatched science sensor?  This seems a mishmash of the two approaches, and you could imagine someone wanting the raw engineering data.

Reply:
I think this needs to be updated now?

Reply:
Thanks. Docs updated, and hopefully reasonably simplified